pkg/broker/broker_test.go

```diff
@@ -143,56 +110,42 @@ func TestServiceDefinition_UserDefinedPlans(t *testing.T) {
 				FieldName: "instances",
 				Type:      JsonTypeString,
 			},
+			{
+				Required:  false,
+				FieldName: "optional_field",
+				Type:      JsonTypeString,
+			},
 		},
 	}
 
 	for tn, tc := range cases {
 		t.Run(tn, func(t *testing.T) {
-			viper.Set(service.UserDefinedPlansProperty(), tc.Value)
-			defer viper.Reset()
+			config := ServiceConfig{CustomPlans: tc.Value}
+			err := service.SetConfig(config)
+			defer service.SetConfig(ServiceConfig{})
 
-			plans, err := service.UserDefinedPlans()
-
-			// Check errors
-			hasErr := err != nil
-			if hasErr != tc.ExpectError {
+			if !reflect.DeepEqual(err, tc.ExpectError) {
 				t.Fatalf("Expected Error? %v, got error: %v", tc.ExpectError, err)
 			}
-
-			// Check IDs
-			if len(plans) != len(tc.PlanIds) {
-				t.Errorf("Expected %d plans, but got %d (%v)", len(tc.PlanIds), len(plans), plans)
-			}
-
-			for _, plan := range plans {
-				if _, ok := tc.PlanIds[plan.ID]; !ok {
-					t.Errorf("Got unexpected plan id %s, expected %+v", plan.ID, tc.PlanIds)
-				}
-			}
 		})
 	}
 }
 
 func TestServiceDefinition_CatalogEntry(t *testing.T) {
 	cases := map[string]struct {
-		UserPlans   interface{}
-		PlanIds     map[string]bool
-		ExpectError bool
+		UserPlans []CustomPlan
+		PlanIds   map[string]bool
 	}{
 		"no-customization": {
-			UserPlans:   nil,
-			PlanIds:     map[string]bool{},
-			ExpectError: false,
+			UserPlans: nil,
+			PlanIds:   map[string]bool{},
 		},
 		"custom-plans": {
-			UserPlans:   `[{"id":"aaa","name":"aaa"},{"id":"bbb","name":"bbb"}]`,
-			PlanIds:     map[string]bool{"aaa": true, "bbb": true},
-			ExpectError: false,
-		},
-		"bad-plan-json": {
-			UserPlans:   `333`,
-			PlanIds:     map[string]bool{},
-			ExpectError: true,
+			UserPlans: []CustomPlan{
```

Comment:
I see the names `UserPlans` and `CustomPlans` used interchangeably. In this example, a customplans struct has a UserPlans field with a CustomPlan type.

UserPlan is a CustomPlan, but not necessarily the other way around.
But regardless, would it be worth converging on a single term? (maybe custom?)